### ISSUE TITLE
Add api-key support.

### ISSUE THREAD

**Issue by skx** [skx/dns-api-go#11]
This project has become a complete support and bandwidth nightmare, so I'm closing the live install.

I've been approached by a couple of users who are willing to pay for them I'll allow access.  Otherwise I think we're done, and users can self-host.

Add (simple) API-key support:

* If X-API-Key is present in the incoming request then
  * Validate it
  * And proceed
* Otherwise return "retired".

**Comment by gulabsisodia-neteffect** [comment]
@skx We want to purchase the key. Can we connect?

**Comment by skx** [comment]
* There was attention when I closed the site.
* The site has been closed for a long time now.
  * Certainly anybody who has been relying on it has had __months__ of broken service.

On that basis trying to bring it back up again, as a commercially supported product/project, seems like a bad idea.   The only thing that would make that useful would be if somebody had hardcoded the `dns-api.org` domain-name in their application - and in that case I'd have to bring it back up for real.  The spam-issue would kill my interest (In that case selling the domain itself would be the better approach.)

I welcome suggestions to the contrary, but with the above in mind I'd have to charge a lot to be interested and that's another reason not to try.
